ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DropSourceCommand.java

```diff
@@ -76,60 +75,44 @@ public DdlCommandResult run(MetaStore metaStore, boolean isValidatePhase) {
         new DropTopicCommand(dataSource.getKsqlTopic().getTopicName());
     metaStore.deleteSource(sourceName);
     dropTopicCommand.run(metaStore, isValidatePhase);
-
     deleteTopicIfNeeded(dataSource, isValidatePhase);
-
     return new DdlCommandResult(true, "Source " + sourceName + " was dropped. "
-                                      + (deleteTopic ? "Topic '"
-                                                    + dataSource.getKsqlTopic().getTopicName()
-                                                    + "' was marked for deletion. Actual deletion "
-                                                    + "and removal from brokers may take some time "
-                                                    + "to complete." : ""));
+        + (deleteTopic ? "Topic '"
+        + dataSource.getKsqlTopic().getTopicName()
+        + "' was marked for deletion. Actual deletion "
+        + "and removal from brokers may take some time "
+        + "to complete." : ""));
   }
 
   private void deleteTopicIfNeeded(StructuredDataSource dataSource, boolean isValidatePhase) {
     if (!isValidatePhase && deleteTopic) {
-
-      executeWithRetries(new Callable<Void>() {
-        @Override
-        public Void call() throws Exception {
+      try {
+        ExecutorWithRetries.execute((Supplier<KafkaFuture<Void>>) () -> {
```

Comment:
Not to sure about the syntax, but is this essentially casting `null` to `Supplier<KafkaFuture<Void>>`? Won't this result in an NPE?

Reply:
Yer, it will. Two options, 

1. `CompletableFuture.completedFuture(null)`. (Assuming you've firsts switched the `execute` method to take `Supplier<? extends Future<T>>`.

1. Add an overload of the `execute` method that just takes a `Supplier<T>`.

Given that we're not doing anything smart / useful with the `Future`, I'd go for option #2.